client/lib/posts/post-list-store.js

```diff
@@ -28,344 +33,350 @@ var _defaultQuery = {
 	perPage: 20
 };
 
-var _activeList = {
-	postIds: [],
-	errors: [],
-	query: clone( _defaultQuery ),
-	page: 0,
-	nextPageHandle: false,
-	isLastPage: false,
-	isFetchingNextPage: false,
-	isFetchingUpdated: false
-};
+let _nextId = 0;
 
-var _nextId = 0;
-var PostListStore;
+module.exports = function( id ) {
+	if ( ! id ) {
+		throw new Error( 'must supply a post-list-store id' );
+	}
 
-function queryPosts( options ) {
-	var query = assign( {}, _defaultQuery, options ),
-		cache = require( './post-list-cache-store' ),
-		list;
+	let _activeList = {
+		postIds: [],
+		errors: [],
+		query: clone( _defaultQuery ),
+		page: 0,
+		nextPageHandle: false,
+		isLastPage: false,
+		isFetchingNextPage: false,
+		isFetchingUpdated: false
+	};
+
+	function queryPosts( options ) {
+		let query = assign( {}, _defaultQuery, options );
+
+		if ( query.siteID && typeof query.siteID === 'string' ) {
+			query.siteID = query.siteID.replace( /::/g, '/' );
+		}
 
-	if ( query.siteID && typeof query.siteID === 'string' ) {
-		query.siteID = query.siteID.replace( /::/g, '/' );
-	}
+		if ( query.status === 'draft,pending' ) {
+			query.orderBy = 'modified';
+		}
 
-	if ( query.status === 'draft,pending' ) {
-		query.orderBy = 'modified';
+		let list = PostListCacheStore.get( query );
+
+		if ( list ) {
+			_activeList = list;
+		} else {
+			_activeList = {
+				id: _nextId,
+				postIds: [],
+				errors: [],
+				query: query,
+				page: 0,
+				isLastPage: false,
+				isFetchingNextPage: false,
+				isFetchingUpdated: false
+			};
+
+			_nextId++;
+		}
 	}
 
-	list = cache.get( query );
-
-	if ( list ) {
-		_activeList = list;
-	} else {
-		_activeList = {
-			id: _nextId,
-			postIds: [],
-			errors: [],
-			query: query,
-			page: 0,
-			isLastPage: false,
-			isFetchingNextPage: false,
-			isFetchingUpdated: false
-		};
-
-		_nextId++;
+	/**
+	 * Remove any keys from the params that are null or undefined
+	 *
+	 * We do this to avoid sending empty values along.
+	 * Returns a new object representing the clean params.
+	 * The original params is unmodified.
+	 *
+	 * @param  {string} params The params to clean
+	 * @return {object} The cleaned params object.
+	 */
+	function cleanParams( params ) {
+		return transform( params, function( result, value, key ) {
+			if ( value != null ) {
+				result[ key ] = value;
+			}
+		}, {} );
 	}
```

Comment:
Like I mentioned above, I'm trying not to touch any of the logic in the base store.  We could do a clean-up sweep on this and some other items @aduth has mentioned earlier in a follow-up branch though.